src/main/java/org/sonar/plugins/github/PullRequestProjectBuilder.java

```diff
@@ -42,10 +47,18 @@ public void build(Context context) {
     if (!gitHubPluginConfiguration.isEnabled()) {
       return;
     }
+    checkMode();
     int pullRequestNumber = gitHubPluginConfiguration.pullRequestNumber();
     pullRequestFacade.init(pullRequestNumber, context.projectReactor().getRoot().getBaseDir());
 
     pullRequestFacade.createOrUpdateSonarQubeStatus(GHCommitState.PENDING, "SonarQube analysis in progress");
   }
 
+  private void checkMode() {
+    if (!settings.getBoolean(CoreProperties.DRY_RUN)) {
+      throw MessageException.of("The GitHub plugin is only intended to be used in preview mode. Please set '" + CoreProperties.ANALYSIS_MODE + "'.");
+    }
```

Comment:
I think so. This is the tricky part here. If I'm correct the only consistent way to check we are in dryRun/preview/issues mode is to use the good old sonar.dryRun property.